java-frontend/src/main/java/org/sonar/java/se/checks/MapComputeIfAbsentOrPresentCheck.java

```diff
@@ -0,0 +1,180 @@
+    return MethodMatcher.create().typeDefinition(TypeCriteria.subtypeOf("java.util.Map")).name(methodName).withAnyParameters();
+  }
+
+  @Override
+  public ProgramState checkPostStatement(CheckerContext context, Tree syntaxNode) {
+    if (syntaxNode.is(Tree.Kind.METHOD_INVOCATION)) {
+      MethodInvocationTree mit = (MethodInvocationTree) syntaxNode;
+      if (MAP_GET.matches(mit)) {
+        ProgramState psBeforeInvocation = context.getNode().programState;
+        ProgramState psAfterInvocation = context.getState();
+
+        SymbolicValue keySV = psBeforeInvocation.peekValue(0);
+        SymbolicValue mapSV = psBeforeInvocation.peekValue(1);
+        SymbolicValue valueSV = psAfterInvocation.peekValue();
+
+        mapGetInvocations.put(mapSV, new MapGetInvocation(valueSV, keySV, mit));
+      }
+    }
+    return super.checkPostStatement(context, syntaxNode);
+  }
+
+  @Override
+  public ProgramState checkPreStatement(CheckerContext context, Tree syntaxNode) {
+    if (syntaxNode.is(Tree.Kind.METHOD_INVOCATION)) {
+      MethodInvocationTree mit = (MethodInvocationTree) syntaxNode;
+      if (MAP_PUT.matches(mit)) {
+        ProgramState ps = context.getState();
+
+        SymbolicValue keySV = ps.peekValue(1);
+        SymbolicValue mapSV = ps.peekValue(2);
+        mapGetInvocations.get(mapSV).stream()
+          .filter(getOnSameMap -> getOnSameMap.withSameKey(keySV))
+          .findAny()
+          .ifPresent(getOnSameMap -> {
+            ObjectConstraint constraint = ps.getConstraint(getOnSameMap.value, ObjectConstraint.class);
+            if (constraint != null) {
+              checkIssues.add(new CheckIssue(context.getNode(), getOnSameMap.mit, mit, getOnSameMap.value, constraint));
+            }
+          });
+      }
+    }
+    return super.checkPreStatement(context, syntaxNode);
+  }
+
+  @Override
+  public void checkEndOfExecution(CheckerContext context) {
+    SECheck check = this;
+    checkIssues.stream().filter(CheckIssue::isOnlyPossibleConstraintOnValueForAllPaths).forEach(issue -> issue.report(context, check));
+  }
+
+  private static class CheckIssue {
+    private final ExplodedGraph.Node node;
+
+    private final MethodInvocationTree getInvocation;
+    private final MethodInvocationTree putInvocation;
+
+    private final SymbolicValue value;
+    private final ObjectConstraint valueConstraint;
+
+    private CheckIssue(Node node, MethodInvocationTree getInvocation, MethodInvocationTree putInvocation, SymbolicValue value, ObjectConstraint valueConstraint) {
+      this.node = node;
+
+      this.getInvocation = getInvocation;
+      this.putInvocation = putInvocation;
+
+      this.value = value;
+      this.valueConstraint = valueConstraint;
+    }
+
+    private boolean isOnlyPossibleConstraintOnValueForAllPaths() {
+      return node.siblings().stream().allMatch(sibling -> valueConstraint.equals(sibling.programState.getConstraint(value, ObjectConstraint.class)));
```

Comment:
no need to go through siblings, we should keep this verification of raising issue only one kind of constraint contained in this check.